mcc/MixedFunctions.c

```diff
@@ -779,3 +795,106 @@ LONG CountLines(struct InstData *data, struct MinList *lines)
+  LONG line_nr = 0;
+  LONG last_line;
+  LONG result = 0;
+  LONG cursor_width = data->CursorWidth == 6 ? TextLength(&data->tmprp, (char *)" ", 1) : data->CursorWidth;
+
+  /* Since this function will be only used in NoWrapMode and
+     in that mode line_nr's will always match the real lines
+     we can find the top (real) line just by traversing the list */
+
+  while(line != NULL && line_nr != data->visual_y - 1)
+  {
+    line = GetNextLine(line);
+    line_nr++;
+  }
+
+  // Calculate the line_nr of the last line displayed
+  last_line = line_nr + data->maxlines - 1;
+
+  /* We found the top line displayed. Now we'll traverse until the
+     last displayed line meanwhile calculating their pixel lengths */
+
+  while(line != NULL && line_nr <= last_line)
+  {
+    result = MAX(result, TextLengthNew(&data->tmprp, line->line.Contents, line->line.Length - 1, data->TabSizePixels));
+    line = GetNextLine(line);
+    line_nr++;
+  }
+
+  return(result + cursor_width);
+}
+///
+/// ScrollIntoView()
+/* ***************************************************************** /
+/  Scrolls the gadget horizontally to make the cursor visible again. /
+/  Do not mistake this with ScrollIntoDisplay() function!            /
+/  (This function is supposed to be used only in NoWrapMode)         /
+/  ******************************************************************/
+void ScrollIntoView(struct InstData *data)
+{
+  struct line_node *line = data->actualline;
+  LONG cursor_width;
+  LONG xmin = 0;
+  LONG xmax;
+
+  if (isFlagClear(data->flags, FLG_HScroll))
+    return;
+
+  xmin  = TextLengthNew(&data->tmprp, line->line.Contents, data->CPos_X, data->TabSizePixels);
+  xmin += FlowSpace(data, line->line.Flow, line->line.Contents);
+
+  // calculate the cursor width
+  // if it is set to 6 then we should find out how the width of the current char is
+  if(data->CursorWidth == 6)
+    cursor_width = TextLength(&data->tmprp, line->line.Contents[data->CPos_X] < ' ' ? (char *)" " : (char *)&line->line.Contents[data->CPos_X], 1);
+  else
+    cursor_width = data->CursorWidth;
+
+  xmax = xmin + cursor_width;
+
+  if (xmin < (LONG)data->xpos)
+  {
+    data->xpos = xmin;
+    DumpText(data, data->visual_y, 0, data->maxlines, FALSE);
+  }
+  else if (xmax > (LONG)(data->xpos + _mwidth(data->object)))
+  {
+    data->xpos = xmax - _mwidth(data->object);
+    DumpText(data, data->visual_y, 0, data->maxlines, FALSE);
+  }
+}
+
+///
+/// MovedIntoDisplay()
+/*---------------------------------------------------------------*
+ *  A function to test if the user moved the cursor into display *
+ *  when it was scrolled out of it                               *
+ *  This function is supposed to work only in NoWrapMode         *
+ *---------------------------------------------------------------*/
+BOOL MovedIntoDisplay(struct InstData *data, struct line_node *oldline)
+{
+    struct line_node *line = GetFirstLine(&data->linelist);
```

Comment:
This function is never used (it was kind'a experimental). I'm completely removing it.